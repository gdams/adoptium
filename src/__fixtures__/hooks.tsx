import {
  Binary,
  ContributorApiResponse,
  Contributor,
  LatestTemurin,
  MarketplaceRelease,
  News,
  ReleaseAsset,
  TemurinReleases,
} from '../hooks';

export const createRandomContributorApiData = (): ContributorApiResponse => ({
  login: 'login_mock',
  id: 0,
  node_id: 'node_id_mock',
  avatar_url: 'avatar_url_mock',
  gravatar_id: 'gravatar_id_mock',
  url: 'url_mock',
  html_url: 'html_url_mock',
  followers_url: 'followers_url_mock',
  following_url: 'following_url_mock',
  gists_url: 'gists_url_mock',
  starred_url: 'starred_url_mock',
  subscriptions_url: 'subscriptions_url_mock',
  organizations_url: 'organizations_url_mock',
  repos_url: 'repos_url_mock',
  events_url: 'events_url_mock',
  received_events_url: 'received_events_url_mock',
  type: 'type_mock',
  site_admin: false,
  contributions: 3,
});

export const createRandomTemurinRelease = (installer, id): ReleaseAsset => ({
  os: 'os_mock',
  architecture: 'arch_mock',
  release_link: new URL('https://release_link_mock'),
  platform_name: `platform_name_mock${id}`,
  release_name: 'release_name_mock',
  release_date: new Date(Date.UTC(2020, 0, 1)),
  binaries: [
    {
      type: 'type_mock',
      link: new URL('https://link_mock'),
      checksum: 'checksum_mock',
      size: 0,
      extension: 'extension_mock',
      installer_link: installer ? new URL('https://installer_link_mock') : undefined,
      installer_checksum: installer ? 'installer_checksum_mock' : undefined,
      installer_size: installer ? 0 : undefined,
      installer_extension: installer ? 'installer_extension_mock' : undefined,
    },
  ],
})

export const createRandomMarketplaceRelease = (installer): MarketplaceRelease => ({
  release_name: 'release_name_mock',
  vendor: 'vendor_mock',
  binary: {
    os: 'os_mock',
    architecture: 'arch_mock',
    distribution: 'distribution_mock',
    image_type: 'type_mock',
    jvm_impl: 'jvm_impl_mock',
    timestamp: new Date(Date.UTC(2020, 0, 1)),
    scm_ref: 'scm_ref_mock',
    openjdk_scm_ref: 'openjdk_scm_ref_mock',
    aqavit_results_link: new URL('https://aqavit_results_link_mock'),
    tck_affidavit_link: new URL('https://tck_affidavit_link_mock'),
    package: {
      name: 'name_mock.tar.gz',
      link: new URL('https://link_mock'),
      sha265sum: 'sha265sum_mock',
      sha256sum_link: new URL('https://sha256sum_link_mock'),
      signature_link: new URL('https://signature_link_mock'),
    },
    installer: installer
      ? [
          {
            name: 'installer_name_mock.msi',
            link: new URL('https://installer_link_mock'),
            sha265sum: 'installer_sha265sum_mock',
            sha256sum_link: new URL('https://installer_sha256sum_link_mock'),
            signature_link: new URL('https://installer_signature_link_mock'),
          },
        ]
      : undefined,
  },
  version: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  }
})

export const mockLatestTemurin = (installer): LatestTemurin => ({
  download_count: 0,
  id: 'id_mock',
  release_link: new URL('https://release_link_mock'),
  release_name: 'release_name_mock',
  release_type: 'release_type_mock',
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  updated_at: new Date(Date.UTC(2020, 0, 1)),
  vendor: 'vendor_mock',
  version_data: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  },
  binaries: [
    {
      os: 'os_mock',
      architecture: 'arch_mock',
      image_type: 'type_mock',
      heap_size: 'heap_size_mock',
      download_count: 0,
      jvm_impl: 'jvm_impl_mock',
      package: {
        name: 'name_mock.tar.gz',
        link: new URL('https://link_mock'),
        checksum: 'sha265sum_mock',
        checksum_link: new URL('https://sha256sum_link_mock'),
        metadata_link: new URL('https://metadata_link_mock'),
        signature_link: new URL('https://signature_link_mock'),
        size: 0,
        download_count: 0,
      },
      installer: installer
      ?
        {
          name: 'installer_name_mock.msi',
          link: new URL('https://installer_link_mock'),
          checksum: 'installer_sha265sum_mock',
          checksum_link: new URL('https://installer_sha256sum_link_mock'),
          metadata_link: new URL('https://installer_metadata_link_mock'),
          signature_link: new URL('https://installer_signature_link_mock'),
          size: 0,
          download_count: 0,
        }
      : undefined,
    }
  ]
})

export const createRandomTemurinReleases = (installer, id): TemurinReleases => ({
  release_name: `release_name_mock_${id}`,
  release_link: new URL('https://release_link_mock'),
  source_url: new URL('https://source_url_mock'),
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  platforms: {
    'platform_mock': {
      assets: [
        {
          os: 'os_mock',
          architecture: 'architecture_mock',
          type: 'type_mock',
          link: new URL('https://link_mock'),
          checksum: `checksum_mock${id}`,
          size: 0,
          extension: 'extension_mock',
          installer_link: installer ? new URL('https://installer_link_mock') : undefined,
          installer_checksum: installer ? 'installer_checksum_mock' : undefined,
          installer_size: installer ? 0 : undefined,
          installer_extension: installer ? 'installer_extension_mock' : undefined,
        },
      ],
    },
  },
})

export const createRandomContributorViewData = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 0,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});

export const createRandomLatestForOSData = (): Binary => ({
  release_name: 'release_name_mock',
  link: new URL('https://link_mock'),
});

export const createRandomNewsAndEventsData = (): News => ({
  news: [
    {
      id: 'id_mock',
      title: 'title_mock',
      body: 'body_mock',
      date: new Date(Date.UTC(2020, 0, 1)),
      link: 'link_mock',
    },
  ],
  events: [
    {
      id: 'id_mock',
      title: 'title_mock',
      infoLink: 'infoLink_mock',
      date: new Date(Date.UTC(2020, 0, 1)),
    },
  ]
});

export const createRandomContributorViewData1Contribution = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 1,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});