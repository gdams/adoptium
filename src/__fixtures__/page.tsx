import {
    MDXPage,
    SingleMDXPage,
    SingleMDXDocPage,
    SiteMetaData
} from '../types';

export const createMDXData = (): {
    allMdx: MDXPage;
} => ({
    allMdx: {
        edges: [
            {
                node: {
                    frontmatter: {
                        title: 'MDX Page title',
                        author: 'pmc',
                        date: '2021-01-01',
                        describtion: 'MDX Page describtion',
                    },
                    fields: {
                        slug: '/mdx/mdx-page-title',
                        postPath: '/mdx/mdx-page-title',
                    },
                    excerpt: 'MDX Page excerpt',
                }
            }
        ]
    }
});

export const createSingleMDXData = (): {
    mdx: SingleMDXPage;
    site: SiteMetaData;
} => ({
    mdx: {
        id: 'mdx-1',
        excerpt: 'MDX Page excerpt',
        frontmatter: {
            title: 'MDX Page title',
            author: 'pmc',
            date: '2021-01-01',
            description: 'MDX Page describtion',
            tags: ['tag1', 'tag2'],
        }
    },
    site: {
        siteMetadata: {
            title: 'Site title',
            siteUrl: 'https://adoptium.net',
            social: {
                twitter: 'twitter',
            }
        }
    }
});

export const createMDXDocsData = (toc): {
    mdx: SingleMDXDocPage;
} => ({
    mdx: {
        id: 'mdx-1',
        fields: {
            slug: '/mdx/mdx-page-title',
        },
        frontmatter: {
            title: 'MDX Doc Page title',
            authors: 'pmc, pmc2',
            toc: toc ? true : false
        },
        tableOfContents: {
            items: [
                {
                    url: '#section1',
                    title: 'Section 1'
                },
                {
                    url: '#section2',
                    title: 'Section 2'
                }
            ],
        }
    }
});
