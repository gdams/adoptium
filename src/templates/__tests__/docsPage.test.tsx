import React from 'react';
import { render } from '@testing-library/react';
import DocsPage, { Head } from '../docsPage';
import { describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe';
import { createMDXDocsData } from '../../__fixtures__/page';

let mockData = createMDXDocsData(false);
const pageContext = {
  relativePath: 'docs/sample/index.md',
}

describe('Docs Template page', () => {
  it('renders correctly', () => {
    const { container } = render(<DocsPage data={mockData} pageContext={pageContext} children={'sample docs'} />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('toc renders correctly', () => {
    let mockData = createMDXDocsData(true);
    const { container } = render(<DocsPage data={mockData} pageContext={pageContext} children={'sample docs'} />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('head renders correctly', () => {
    const { container } = render(<Head data={mockData} />);
    // eslint-disable-next-line
    const title = container.querySelector('title');
    expect(title?.textContent).toEqual('MDX Doc Page title | Adoptium');
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<DocsPage data={mockData} pageContext={pageContext} children={'sample docs'} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});