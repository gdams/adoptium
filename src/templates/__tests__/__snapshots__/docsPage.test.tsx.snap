// Vitest Snapshot v1

exports[`Docs Template page > installation renders correctly 1`] = `
<main>
  <section
    class="py-5 px-3"
  >
    <div
      class="container-mdx row"
    >
      <div
        class="col-lg-3 hide-on-mobile"
      />
      <div
        class="col-lg-6 col-md-12"
      >
        <h1
          class="pb-4 fw-light text-center"
        >
          MDX Doc Page title
        </h1>
        <section
          class="adopt-demo-container hide-on-mobile my-5"
        >
          <div
            class="adopt-demo mx-auto"
          >
            <div
              class="install"
            >
              <div
                class="react-tabs"
                data-rttabs="true"
              >
                <div
                  class="install__header"
                >
                  <div
                    class="install__header-circles"
                  >
                    <div
                      class="install__header-grey-circle red"
                    />
                    <div
                      class="install__header-grey-circle yellow"
                    />
                    <div
                      class="install__header-grey-circle green"
                    />
                  </div>
                  <div
                    class="install__header-text"
                  >
                    bash
                  </div>
                </div>
                <ul
                  class="react-tabs__tab-list"
                  role="tablist"
                >
                  <li
                    aria-controls="panel:r0:0"
                    aria-disabled="false"
                    aria-selected="true"
                    class="react-tabs__tab react-tabs__tab--selected"
                    data-rttab="true"
                    id="tab:r0:0"
                    role="tab"
                    tabindex="0"
                  >
                    Windows (Winget)
                  </li>
                  <li
                    aria-controls="panel:r0:1"
                    aria-disabled="false"
                    aria-selected="false"
                    class="react-tabs__tab"
                    data-rttab="true"
                    id="tab:r0:1"
                    role="tab"
                  >
                    macOS (Homebrew)
                  </li>
                  <li
                    aria-controls="panel:r0:2"
                    aria-disabled="false"
                    aria-selected="false"
                    class="react-tabs__tab"
                    data-rttab="true"
                    id="tab:r0:2"
                    role="tab"
                  >
                    Linux (RPM/DEB)
                  </li>
                </ul>
                <div
                  aria-labelledby="tab:r0:0"
                  class="react-tabs__tab-panel react-tabs__tab-panel--selected"
                  id="panel:r0:0"
                  role="tabpanel"
                >
                  <div>
                    <pre
                      class="shell-box no-highlight"
                    >
                      <div
                        class="shell-box-top"
                      />
                      <code
                        class="shell-box-code"
                      >
                        <span
                          class="install__text__no-select"
                        >
                          # Install the latest version
                        </span>
                        <br />
                        <span
                          class="install__text__no-select"
                        >
                          $
                        </span>
                        <span
                          class="install-text-command"
                        >
                          winget install 
                        </span>
                        EclipseAdoptium.Temurin.17.JDK
                      </code>
                    </pre>
                    <br />
                    <pre
                      class="shell-box no-highlight"
                    >
                      <div
                        class="shell-box-top"
                      />
                      <code
                        class="shell-box-code"
                      >
                        <span
                          class="install__text__no-select"
                        >
                          # Install a different version
                        </span>
                        <br />
                        <span
                          class="install__text__no-select"
                        >
                          $
                        </span>
                        <span
                          class="install-text-command"
                        >
                          winget install 
                        </span>
                        EclipseAdoptium.Temurin.11.JDK
                      </code>
                    </pre>
                    <br />
                    <pre
                      class="shell-box no-highlight"
                    >
                      <div
                        class="shell-box-top"
                      />
                      <code
                        class="shell-box-code"
                      >
                        <span
                          class="install__text__no-select"
                        >
                          # Upgrade the Winget package
                        </span>
                        <br />
                        <span
                          class="install__text__no-select"
                        >
                          $
                        </span>
                        <span
                          class="install-text-command"
                        >
                          winget upgrade 
                        </span>
                        EclipseAdoptium.Temurin.17.JDK
                      </code>
                    </pre>
                    <br />
                    <pre
                      class="shell-box no-highlight"
                    >
                      <div
                        class="shell-box-top"
                      />
                      <code
                        class="shell-box-code"
                      >
                        <span
                          class="install__text__no-select"
                        >
                          # Uninstall the Winget package
                        </span>
                        <br />
                        <span
                          class="install__text__no-select"
                        >
                          $
                        </span>
                        <span
                          class="install-text-command"
                        >
                          winget uninstall 
                        </span>
                        EclipseAdoptium.Temurin.17.JDK
                      </code>
                    </pre>
                    <a
                      class="install__docs-button"
                      href="https://docs.microsoft.com/en-us/windows/package-manager/winget/"
                    >
                      Read documentation
                    </a>
                  </div>
                </div>
                <div
                  aria-labelledby="tab:r0:1"
                  class="react-tabs__tab-panel"
                  id="panel:r0:1"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="tab:r0:2"
                  class="react-tabs__tab-panel"
                  id="panel:r0:2"
                  role="tabpanel"
                />
              </div>
            </div>
          </div>
        </section>
        sample docs
        <hr
          class="m-5"
        />
        <div
          class="_authorList_b2bb3b"
        >
          DOCUMENTATION AUTHORS
          <ul>
            <li>
              <a
                aria-label="pmc"
                class="_link_09add8 author-icon"
                href="https://github.com/pmc"
                rel="noopener noreferrer"
                style="margin-left: 0px;"
                target="_blank"
              >
                <img
                  alt="pmc"
                  src="https://github.com/pmc.png?size=60"
                />
              </a>
            </li>
            <li>
              <a
                aria-label="pmc2"
                class="_link_09add8 author-icon"
                href="https://github.com/pmc2"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt=" pmc2"
                  src="https://github.com/pmc2.png?size=60"
                />
              </a>
            </li>
          </ul>
        </div>
        <div
          class="mx-auto pt-5"
        >
          <span
            class="fw-bold"
          >
            Help us make these docs great!
          </span>
          <br />
          <span>
            All Adoptium docs are open source. See something that's wrong or unclear?
          </span>
          <br />
          <a
            class="btn btn-secondary mt-2"
            href="https://github.com/adoptium/adoptium.net/edit/main/content/mdx-docs/docs/sample/index.md"
            rel="noopener noreferrer"
            style="border-color: #0869da; background-color: rgb(246, 248, 250);"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="18"
              stroke="currentColor"
              stroke-width="0"
              style="margin-right: .3em;"
              viewBox="0 0 512 512"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 376.6V206.8c0-46.4-3.7-70.8-22.1-94C374 87.7 348.9 77 310 77h-22V16l-96 96 96 96v-65h22c24.6 0 29 3.6 33.8 9.6 5.5 6.9 8.2 19.1 8.2 54.2v169.8c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4zM384 472c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zM128 48c-35.3 0-64 28.7-64 64 0 23.7 12.9 44.3 32 55.4v209.2c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4V167.4c19.1-11.1 32-31.7 32-55.4 0-35.3-28.7-64-64-64zm0 424c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm0-320c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40z"
              />
            </svg>
            Edit this page
          </a>
        </div>
      </div>
      <div
        class="col-lg-3 hide-on-mobile"
      />
    </div>
  </section>
</main>
`;

exports[`Docs Template page > renders correctly 1`] = `
<main>
  <section
    class="py-5 px-3"
  >
    <div
      class="container-mdx row"
    >
      <div
        class="col-lg-3 hide-on-mobile"
      />
      <div
        class="col-lg-6 col-md-12"
      >
        <h1
          class="pb-4 fw-light text-center"
        >
          MDX Doc Page title
        </h1>
        sample docs
        <hr
          class="m-5"
        />
        <div
          class="_authorList_b2bb3b"
        >
          DOCUMENTATION AUTHORS
          <ul>
            <li>
              <a
                aria-label="pmc"
                class="_link_09add8 author-icon"
                href="https://github.com/pmc"
                rel="noopener noreferrer"
                style="margin-left: 0px;"
                target="_blank"
              >
                <img
                  alt="pmc"
                  src="https://github.com/pmc.png?size=60"
                />
              </a>
            </li>
            <li>
              <a
                aria-label="pmc2"
                class="_link_09add8 author-icon"
                href="https://github.com/pmc2"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt=" pmc2"
                  src="https://github.com/pmc2.png?size=60"
                />
              </a>
            </li>
          </ul>
        </div>
        <div
          class="mx-auto pt-5"
        >
          <span
            class="fw-bold"
          >
            Help us make these docs great!
          </span>
          <br />
          <span>
            All Adoptium docs are open source. See something that's wrong or unclear?
          </span>
          <br />
          <a
            class="btn btn-secondary mt-2"
            href="https://github.com/adoptium/adoptium.net/edit/main/content/mdx-docs/docs/sample/index.md"
            rel="noopener noreferrer"
            style="border-color: #0869da; background-color: rgb(246, 248, 250);"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="18"
              stroke="currentColor"
              stroke-width="0"
              style="margin-right: .3em;"
              viewBox="0 0 512 512"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 376.6V206.8c0-46.4-3.7-70.8-22.1-94C374 87.7 348.9 77 310 77h-22V16l-96 96 96 96v-65h22c24.6 0 29 3.6 33.8 9.6 5.5 6.9 8.2 19.1 8.2 54.2v169.8c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4zM384 472c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zM128 48c-35.3 0-64 28.7-64 64 0 23.7 12.9 44.3 32 55.4v209.2c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4V167.4c19.1-11.1 32-31.7 32-55.4 0-35.3-28.7-64-64-64zm0 424c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm0-320c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40z"
              />
            </svg>
            Edit this page
          </a>
        </div>
      </div>
      <div
        class="col-lg-3 hide-on-mobile"
      />
    </div>
  </section>
</main>
`;

exports[`Docs Template page > toc renders correctly 1`] = `
<main>
  <section
    class="py-5 px-3"
  >
    <div
      class="container-mdx row"
    >
      <div
        class="col-lg-3 hide-on-mobile"
      />
      <div
        class="col-lg-6 col-md-12"
      >
        <h1
          class="pb-4 fw-light text-center"
        >
          MDX Doc Page title
        </h1>
        <details
          class="p-3 my-3 bg-grey"
        >
          <summary
            class="lead"
          >
            Table of Contents
          </summary>
          <ul>
            <li>
              <a
                href="#section1"
              >
                Section 1
              </a>
            </li>
            <li>
              <a
                href="#section2"
              >
                Section 2
              </a>
            </li>
          </ul>
        </details>
        sample docs
        <hr
          class="m-5"
        />
        <div
          class="_authorList_b2bb3b"
        >
          DOCUMENTATION AUTHORS
          <ul>
            <li>
              <a
                aria-label="pmc"
                class="_link_09add8 author-icon"
                href="https://github.com/pmc"
                rel="noopener noreferrer"
                style="margin-left: 0px;"
                target="_blank"
              >
                <img
                  alt="pmc"
                  src="https://github.com/pmc.png?size=60"
                />
              </a>
            </li>
            <li>
              <a
                aria-label="pmc2"
                class="_link_09add8 author-icon"
                href="https://github.com/pmc2"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt=" pmc2"
                  src="https://github.com/pmc2.png?size=60"
                />
              </a>
            </li>
          </ul>
        </div>
        <div
          class="mx-auto pt-5"
        >
          <span
            class="fw-bold"
          >
            Help us make these docs great!
          </span>
          <br />
          <span>
            All Adoptium docs are open source. See something that's wrong or unclear?
          </span>
          <br />
          <a
            class="btn btn-secondary mt-2"
            href="https://github.com/adoptium/adoptium.net/edit/main/content/mdx-docs/docs/sample/index.md"
            rel="noopener noreferrer"
            style="border-color: #0869da; background-color: rgb(246, 248, 250);"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="18"
              stroke="currentColor"
              stroke-width="0"
              style="margin-right: .3em;"
              viewBox="0 0 512 512"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 376.6V206.8c0-46.4-3.7-70.8-22.1-94C374 87.7 348.9 77 310 77h-22V16l-96 96 96 96v-65h22c24.6 0 29 3.6 33.8 9.6 5.5 6.9 8.2 19.1 8.2 54.2v169.8c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4zM384 472c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zM128 48c-35.3 0-64 28.7-64 64 0 23.7 12.9 44.3 32 55.4v209.2c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4V167.4c19.1-11.1 32-31.7 32-55.4 0-35.3-28.7-64-64-64zm0 424c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm0-320c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40z"
              />
            </svg>
            Edit this page
          </a>
        </div>
      </div>
      <div
        class="col-lg-3 hide-on-mobile"
      />
    </div>
  </section>
</main>
`;
